src/test/java/org/jfrog/buildinfo/integration/ArtifactoryPluginITest.java

```diff
@@ -167,7 +179,7 @@ private void runProject(String projectName) throws VerificationException, IOExce
             verifier.setEnvironmentVariable("MAVEN_OPTS", "-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=5005");
         }
         verifier.getVerifierProperties().put("use.mavenRepoLocal", "false");
-        verifier.executeGoals(Lists.newArrayList("clean", "deploy", "-Dartifactory.plugin.version=" + getPluginVersion()));
+        verifier.executeGoals(Lists.newArrayList("clean", "deploy", "-Dartifactory.plugin.version=" + getPluginVersion(), "-Dpassword2=123", "-s", "settings.xml"));
```

Comment:
I couldn't find where the "password2" is used. Shouldn't be in the "badProps" list?

I suggest refactoring this logic a little bit:
* Add 1 system property expected to be filtered
* Add 1 system property expected to not be filtered
* Add 1 environment variable expected to be filtered
* Add 1 environment variable expected to not be filtered

In the getAndAssert build search for the ones expected to appear in the build-info and make sure the ones expected to be filtered out don't appear.